pkg/actuators/machine/vm.go

```diff
@@ -265,6 +348,15 @@ func createVM(mscp *machineScope, userData []byte) (*nutanixClientV3.VMIntentRes
 			UUID: mscp.providerSpec.Cluster.UUID,
 		}
 
+		// Set boot_type if configured in the machine's providerSpec
+		if mscp.providerSpec.BootType == machinev1.NutanixUEFIBoot {
+			vmSpec.Resources.BootConfig.BootType = utils.StringPtr("UEFI")
+		} else if mscp.providerSpec.BootType == machinev1.NutanixSecureBoot {
+			vmSpec.Resources.BootConfig.BootType = utils.StringPtr("SECURE_BOOT")
+		} else {
+			// The vm uses the default "LEGACY" boot type
```

Comment:
Should we not set this explicitly rather than relying on Nutanix to default it? What would happen if Nutanix changed the default on their platform, this could break users

Reply:
Make sense. Will fix this.

Reply:
Fixed.